include/glow/Backends/DeviceManager.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_DEVICEMANAGER_H
+#define GLOW_BACKENDS_DEVICEMANAGER_H
+
+#include "glow/Backends/Backend.h"
+#include "glow/Backends/CompiledFunction.h"
+#include "glow/Graph/Context.h"
+#include "glow/Graph/Graph.h"
+#include "glow/Runtime/RuntimeTypes.h"
+
+#include <functional>
+#include <map>
+#include <string>
+
+namespace glow {
+
+using ReadyCBTy = std::function<void(const Module *, runtime::ResultCode)>;
```

Comment:
I think we can drop Module * as a parameter to the CB, If it's purpose is to be an identifier that can be captured in the lambda the caller passes.

Reply:
There's two schools of thought on this PR about whether to rely entirely on the lambda capture or not. Since we have the RunIdentifier for runFunction I think we should have something for AddNetwork as well; perhaps we should bring back DeviceNetworkId?

Reply:
I think I'd prefer to keep the Module * over bringing back DeviceNetworkId, since the function name uniquely identifies the function to the Device Manager already a second identifier seems redundant. I'm good with keeping Module * so we do not have to be reliant on the lambda capture though.